### ISSUE TITLE
[BUG] Segmented Control arrow key behavior does not work when the host component uses shadow DOM encapsulation

### ISSUE THREAD

**Issue by Loujuu** [kirbydesign/designsystem#4209]
- [ ] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
Tab behavior in segmented control does not work when the host component is inside a shadow DOM. If you press Enter / Space, you will get an error in the log that suggests focus is not set on ion-segment-button, but remains on ion-segment.

If you delete the encapsulation from the Modal Content component, then the tabs work as intended.

Related issue: https://github.com/kirbydesign/designsystem/issues/3687 and https://github.com/ionic-team/ionic-framework/issues/30022

## Describe how to reproduce the bug
Look at this example: https://stackblitz.com/edit/stackblitz-starters-ybnkkhy9?file=src%2Fmain.ts

## Please complete the following information:
- OS: [e.g. Windows, MacOS, iOS]
- Browser [e.g. Chrome, Safari]
- Browser version [e.g. 22]

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/develop/.github/SUPPORT.md#team-kirby); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by RasmusKjeldgaard** [comment]
Both the issue for pressing enter and navigating segment buttons with arrow keys occur because `ion-segment` uses `document.activeElement` to manage roving tabindex  [here](https://github.com/ionic-team/ionic-framework/blob/442e3e983107a69cea4fb5587fb33da718eee8a3/core/src/components/segment/segment.tsx#L684) and [here](https://github.com/ionic-team/ionic-framework/blob/442e3e983107a69cea4fb5587fb33da718eee8a3/core/src/components/segment/segment.tsx#L643).

The intention of the shadow root is to fully encapsulate its contents, so when a segment-button inside the shadow host has focus, `document.activeElement` points to the host, not the actual button that has focus, which ends up leading to a mismatch in the ion-segment logic for handling arrow keys.

While unfortunate that the focus is handled this way, I think the best we can do for now is to ensure that segmented control is always in the light dom, either by using a `<slot>` in combination with the declarative modal like so 

```
<kirby-modal [size]="size" flavor="modal" trigger="open-modal">
  <ng-template>
      <my-shadow-root-host>
          <kirby-segmented-control></kirby-segmented-control>
     </my-shadow-root-host>
  </ng-template>
</kirby-modal>
```
or by composing the component provided to the modal controller into more sub-components that each, except for the segmented-control host, implement `ViewEncapsulation.ShadowDom`.

I'll leave this bug-report open, so we get back to it when converting the segmented-control to a web component of our own making.

**Comment by CuddlyKalja** [comment]
A similar bug has been found with Shadowdoms and Modals.
You cannot tab through selectables in a modal, as long as it is inside a shadowdom.
A SB that showcases this can be seen here:
https://stackblitz.com/edit/stackblitz-starters-7x7tntlx?file=src%2Fmain.ts,src%2Findex.html

**Comment by RasmusKjeldgaard** [comment]
Closing this issue automatically because it has been open for more than 6 months and is not currently in progress. If this is still relevant, please feel free to reopen it or open a new issue with updated details. Thank you!
